scripts/staging/fm_binclass.dml

```diff
@@ -19,32 +19,89 @@
 #
 #-------------------------------------------------------------
 
-/*
+
+/**
  * Factorization Machines for binary classification.
  */
 
 # Imports
 source("staging/fm.dml") as fm
 source("nn/layers/sigmoid.dml") as sigmoid
 source("nn/layers/log_loss.dml") as log_loss
+source("nn/optim/adam.dml") as adam
 
 train = function(matrix[double] X)
-    return (matrix[double] out) {
+    return (matrix[double] Y, matrix[double] X_val, matrix[double] Y_val) {
    /*
     * Trains a FM model.
     */
+    
+    # 1.initialize fm core
+    [w0, W, V] = fm::init(d, k);
+    
+    # 2.initialize adam optimizer
+    [mw0, vw0] = adam::init(w0);
+    [mW, vW] = adam::init(W);
+    [mV, vV] = adam::init(V);
+    
+    # 3.Send inputs through fm::forward
+    y = fm::forward(X, w0, W, V);  
+    
+    # 4.Send the above result through sigmoid::forward
+    sfy = sigmoid::forward(y);
+    
+    # 5.Send the above result through log_loss::forward
+    lfy = log_loss::forward(sfy);
+    
+    # 6.Send the result of sigmoid::forward and the correct labels y to log_loss::backward
+    lby = log_loss::backward(sfy, y);
+    
+    # 7.Send the above result through sigmoid::backward
+    sby = sigmoid::backward(lby);
+    
+    # 8.Send the above result through fm::backward
+    fb = fm::backward(sby);
```

Comment:
rename `sby` -> `dy` from above, and also supply `X, w0, W, V` as given in `fm::forward`.  this will return derivatives [here](https://github.com/apache/systemml/pull/526/files#diff-bccc3e0d67df059416a3a9be773a635dR53).